src/util/redisConnector.js

```diff
@@ -0,0 +1,103 @@
+const Redis = require('ioredis');
+const { RedisError } = require('../v0/util/errorTypes');
+const log = require('../logger');
+const stats = require('./stats');
+const { resolve } = require('path');
+
+const timeoutPromise = new Promise((resolve, reject) => {
+  setTimeout(
+    () => resolve(),
+    50
+  );
+});
+
+const RedisDB = {
+  init() {
+    this.host = process.env.REDIS_HOST || 'localhost';
+    this.port = parseInt(process.env.REDIS_PORT, 10) || 6379;
+    this.password = process.env.REDIS_PASSWORD;
+    this.maxRetries = parseInt(process.env.REDIS_MAX_RETRIES || 30, 10);
+    this.timeAfterRetry = parseInt(process.env.REDIS_TIME_AFTER_RETRY_IN_MS || 10, 10);
+    this.client = new Redis({
+      host: this.host,
+      port: this.port,
+      password: this.password,
+      enableReadyCheck: true,
+      retryStrategy: (times) => {
+        if (times <= this.maxRetries) {
+          return 10 + times * this.timeAfterRetry;
+        }
+        log.error(`Redis is down at ${this.host}:${this.port}`);
+        return false; // stop retrying
+      },
+    });
+    this.client.on('ready', (res) => {
+      stats.increment('redis_ready', {
+        timestamp: Date.now(),
+      });
+      log.info(`Connected to redis at ${this.host}:${this.port}`);
+    });
+  },
+
+  /**
+   * Used to get value from redis depending on the key and the expected value type
+   * @param {*} key key for which value needs to be extracted
+   * @param {*} isJsonExpected false if fetched value can not be json
+   * @returns value which can be json or string or number
+   *
+   */
+  async getVal(key, isJsonExpected = true) {
+    try {
+      if (!this.client) {
+        this.init();
```

Comment:
is this.client.connect function call required here?

Reply:
It will be called in case redis client is not connected. This is useful to keep trying connection with redis in case of failure.